schedule/yast/autoyast_multi_btrfs.yaml

```diff
@@ -26,4 +29,109 @@ test_data:
         - /dev/mapper/cr_vdd1
   crypttab:
     num_devices_encrypted: 1
-  !include: test_data/yast/encryption/default_enc.yaml
\ No newline at end of file
+  !include: test_data/yast/encryption/default_enc.yaml
+  profile:
+    partitioning:
+      - drive:
+          unique_by: 'device'
+          device: /dev/vda
+          disklabel: gpt
+          partitions:
+            - partition:
+                unique_by: partition_nr
+                partition_nr: 1
+            - partition:
+                unique_by: partition_nr
+                partition_nr: 2
+            - partition:
+                unique_by: partition_nr
+                partition_nr: 3
+                filesystem: swap
+                mount: swap
+          type: CT_DISK
+      - drive:
+          unique_by: 'device'
+          device: /dev/vdb
+          disklabel: gpt
+          partitions:
+            - partition:
+                unique_by: partition_nr
+                partition_nr: 1
+          type: CT_DISK
+      - drive:
+          unique_by: 'device'
+          device: /dev/vdc
+          disklabel: none
+          partitions:
+            - partition:
+                unique_by: create
+                create: false
+          type: CT_DISK
+      - drive:
+          unique_by: 'device'
+          device: /dev/vdd
+          disklabel: gpt
+          partitions:
+            - partition: 
+                unique_by: partition_nr
+                partition_nr: 1
+                crypt_method: luks1
+          type: CT_DISK
+      - drive:
+          unique_by: 'partitions/partition[0]/label'
```

Comment:
Feels like we still make it to complex to handle what we actually need. I had in mind that we can have some special field how to check if node is there, but here I can see that we have even some expression in `unique_by` and not just field name like we have in other examples.
In the end all we need is to find correct disk and check some properties in it, so we could use `unique_by` as special key, but we also could simplify it significantly, by just saying I need drive with `/` mount point. So we basically use 2 fields to find drive, either it's just name, or mount point(s) of the partitions on the disk.
With such approach we could at least separate part of data which we use to search and other parts to validate.
Otherwise I would really say that we should form monster xpath and validate whole thing at once. That's the only way to keep same structure as xml. Errors handling would be tricky in this case as we've discussed.

Reply:
Makes sense, I'm not expert in xpath so I guess what you are referring is something like @b10n1k mentioned below about to access the internal search field with some wild card to avoid indexes or long path...that would allow to have that kind of search based in two fields. I have to check it.